module/Finna/src/Finna/View/Helper/Root/RecordImage.php

```diff
@@ -187,6 +189,39 @@ public function getMasterImageWithInfo(
         return compact('url', 'pdf');
     }
 
+    /**
+     * Returns high resolution data with download links
+     *
+     * @param int    $index  Record image data
+     * @param string $source Record source
+     *
+     * @return mixed
+     */
+    public function getHighResolutionDownloadData(
+        $index = null, $source = DEFAULT_SEARCH_BACKEND
+    ) {
+        $data = $this->record->getHighResolutionData($index, $source);
+        if (!$data) {
+            return false;
+        }
+
+        $urlHelper = $this->getView()->plugin('url');
+        foreach ($data as $i => &$images) {
+            foreach ($images as $size => &$links) {
+                if ($size === 'resourceID') {
+                    continue;
+                }
+                foreach ($links as $format => &$values) {
+                    $values['url'] = $urlHelper('cover-download') . '?' .
```

Comment:
URL:t olisi ehkä myös parempi luoda templaatissa. Ja tähän tapaan:
`$this->url('cover-download', [], ['query' => $values['params']);`

Reply:
urli toteutetaan templaatissa recordimage.php sijasta.